src/inabox/inabox-viewport.js

```diff
@@ -133,10 +133,21 @@ class InaboxViewportImpl {
     /** @private @const {!Observable<!../service/viewport/viewport-interface.ViewportResizedEventDef>} */
     this.resizeObservable_ = new Observable();
 
-    this.binding_.connect();
     this.binding_.onScroll(this.scroll_.bind(this));
     this.binding_.onResize(this.resize_.bind(this));
 
+    /** @private {boolean} */
+    this.visible_ = false;
+    this.ampdoc.onVisibilityChanged(this.updateVisibility_.bind(this));
+    this.updateVisibility_();
+
+    // Workaround for Safari not firing visibilityChange when the page is
+    // unloaded (https://bugs.webkit.org/show_bug.cgi?id=151234).
+    const unloadListener = win.addEventListener('pagehide', () => {
```

Comment:
@dvoytenko is `ampdoc` not handling this event?

Reply:
https://github.com/ampproject/amphtml/blob/1c5d54b5e3e4a2c108a558fa66b9cc22ca61058e/src/utils/document-visibility.js#L87-L93

Reply:
Not to my knowledge at this time. This is a newer event I believe. Not sure how it interops with `visibilityChange` besides unloading nuance. But generally it sounds possible given that the new page visibility events are more clear on the spec side for unload/freeze/terminate lifecycle. I'd be totally ok if we decided to handle it on the `ampdoc` side.

Reply:
SG. @zombifier to not block your work here, please add a TODO that this event handling can be moved to ampdoc later.

Reply:
Done.